internal/validitywindow/validitywindow.go

```diff
@@ -179,3 +182,16 @@ func (v *TimeValidityWindow[T]) isRepeat(
 func (v *TimeValidityWindow[T]) calculateOldestAllowed(timestamp int64) int64 {
 	return max(0, timestamp-v.getTimeValidityWindow(timestamp))
 }
+
+func VerifyTimestamp(containerTimestamp int64, executionTimestamp int64, divisor int64, validityWindow int64) error {
+	switch {
+	case containerTimestamp%divisor != 0:
+		return fmt.Errorf("%w: timestamp (%d) %% divisor (%d) != 0", ErrMisalignedTime, containerTimestamp, divisor)
+	case containerTimestamp < executionTimestamp: // expiry: 100 block: 110
+		return fmt.Errorf("%w: timestamp (%d) < block timestamp (%d)", ErrTimestampExpired, containerTimestamp, executionTimestamp)
+	case containerTimestamp > executionTimestamp+validityWindow: // expiry: 100 block 10
```

Comment:
These were in the old code but I feel like we should delete these comments as part of this diff anyways

Reply:
I find these helpful, would prefer not to delete. Would you prefer they be moved in some way?

Reply:
IMO it's unnecessary to comment to explain how `<` and `>` work, but if you find it helpful feel free free to leave it in